main.ts

```diff
@@ -101,6 +104,14 @@ export default class VimrcPlugin extends Plugin {
 		})
 	}
 
+	registerFileOpenEvent(){
+		// Record the position of selections in each file opened
+		let cmEditor = this.getCodeMirror(this.getActiveView());
+		CodeMirror.on(cmEditor, "cursorActivity", async (cm: any) => {

```

Comment:
Are you sure this happens just once on each CM instance? I'm not sure at all that Obsidian guarantees that this doesn't happen an infinite number of times on the same editor instance.
Maybe `on('codemirror', ...)` is a safer place to put this?

Reply:
with `on('codemirror', ...)` couldn't get it to work